### PR TITLE
Add Organization Name to Project Options

### PR DESCRIPTION
**Pull request by sebastianv1**
Add an attribute to project options for populating the `ORGANIZATIONNAME` attribute in the main project group. This is generally used in the new file template header.

### CODE REVIEW

File: examples/integration/xcodeproj_targets.bzl [hunk 1]
```diff
@@ -15,6 +15,7 @@ PROJECT_OPTIONS = project_options(
     indent_width = 3,
     tab_width = 2,
     uses_tabs = True,
+    organization_name = "BB",
```
Comment (brentleyjones) [inline examples/integration/xcodeproj_targets.bzl:18]:
```suggestion
    organization_name = "BB",
    tab_width = 2,
    uses_tabs = True,
```

File: tools/generator/src/DTO/Project.swift [hunk 2]
```diff
@@ -4,17 +4,20 @@ struct Project: Equatable {
         let indentWidth: UInt?
         let tabWidth: UInt?
         let usesTabs: Bool?
+        let organizationName: String?
```
Comment (brentleyjones) [inline tools/generator/src/DTO/Project.swift:7]:
```suggestion
        let organizationName: String?
        let tabWidth: UInt?
        let usesTabs: Bool?
```

File: tools/generator/src/DTO/Project.swift [hunk 3]
```diff
@@ -4,17 +4,20 @@ struct Project: Equatable {
...
 
         init(
             developmentRegion: String = "en",
             indentWidth: UInt? = nil,
             tabWidth: UInt? = nil,
-            usesTabs: Bool? = nil
+            usesTabs: Bool? = nil,
+            organizationName: String? = nil
```
Comment (brentleyjones) [inline tools/generator/src/DTO/Project.swift:14]: Can you move this above `tabWidth`?

File: tools/generator/src/DTO/Project.swift [hunk 4]
```diff
@@ -4,17 +4,20 @@ struct Project: Equatable {
...
         ) {
             self.developmentRegion = developmentRegion
             self.indentWidth = indentWidth
             self.tabWidth = tabWidth
             self.usesTabs = usesTabs
+            self.organizationName = organizationName
```
Comment (brentleyjones) [inline tools/generator/src/DTO/Project.swift:20]:
```suggestion
            self.organizationName = organizationName
            self.tabWidth = tabWidth
            self.usesTabs = usesTabs
```

File: tools/generator/src/DTO/Project.swift [hunk 5]
```diff
@@ -122,6 +125,7 @@ extension Project.Options: Decodable {
         case indentWidth = "i"
         case tabWidth = "t"
         case usesTabs = "u"
+        case organizationName = "o"
```
Comment (brentleyjones) [inline tools/generator/src/DTO/Project.swift:128]:
```suggestion
        case organizationName = "o"
        case tabWidth = "t"
        case usesTabs = "u"
```

File: xcodeproj/internal/project_options.bzl [hunk 6]
```diff
@@ -5,7 +5,8 @@ def project_options(
         development_region = "en",
         indent_width = None,
         tab_width = None,
-        uses_tabs = None):
+        uses_tabs = None,
+        organization_name = None):
```
Comment (brentleyjones) [inline xcodeproj/internal/project_options.bzl:9]: Can you move this above `tab_width`?

File: xcodeproj/internal/project_options.bzl [hunk 7]
```diff
@@ -14,6 +15,7 @@ def project_options(
         indent_width: Optional. The number of spaces to use for indentation.
         tab_width: Optional. The number of spaces to use for tabs.
         uses_tabs: Optional. Whether to use tabs for indentation.
+        organization_name: Optional. Populates the ORGANIZATIONNAME attribute for the project.
```
Comment (brentleyjones) [inline xcodeproj/internal/project_options.bzl:18]:
```suggestion
        organization_name: Optional. Populates the `ORGANIZATIONNAME` attribute
            for the project.
        tab_width: Optional. The number of spaces to use for tabs.
        uses_tabs: Optional. Whether to use tabs for indentation.
```

File: xcodeproj/internal/project_options.bzl [hunk 8]
```diff
@@ -31,6 +33,8 @@ def project_options(
         d["tab_width"] = str(tab_width)
     if uses_tabs != None:
         d["uses_tabs"] = "1" if uses_tabs else "0"
+    if organization_name != None:
+        d["organization_name"] = organization_name
```
Comment (brentleyjones) [inline xcodeproj/internal/project_options.bzl:37]: Can you move this above `tab_width`? Also, change to `if organization_name:`.

File: xcodeproj/internal/project_options.bzl [hunk 9]
```diff
@@ -61,4 +65,8 @@ def project_options_to_dto(project_options):
     if uses_tabs:
         dto["u"] = uses_tabs == "1"
 
+    organization_name = project_options.get("organization_name")
+    if organization_name:
+        dto["o"] = organization_name
+
```
Comment (brentleyjones) [inline xcodeproj/internal/project_options.bzl:71]: Can you move this above `tab_width`?

**Review by brentleyjones — APPROVED** [review]
Thanks!

You'll need to update the docs as well after these changes. You can do that with `./docs/update_docs.sh`.

File: tools/generator/src/DTO/Project.swift (hunk 3 above)
Comment (sebastianv1) [reply tools/generator/src/DTO/Project.swift:14]: Will do!

**Comment by sebastianv1** [comment]
Believe I covered all the reorders

File: tools/generator/src/DTO/Project.swift [hunk 10]
```diff
@@ -131,6 +135,7 @@ extension Project.Options: Decodable {
             .decodeIfPresent(String.self, forKey: .developmentRegion) ?? "en"
         indentWidth = try container
             .decodeIfPresent(UInt.self, forKey: .indentWidth)
+        organizationName = try container.decodeIfPresent(String.self, forKey: .organizationName)
```
Comment (brentleyjones) [inline tools/generator/src/DTO/Project.swift:138]:
```suggestion
        organizationName = try container
            .decodeIfPresent(String.self, forKey: .organizationName)
```

File: xcodeproj/internal/project_options.bzl [hunk 11]
```diff
@@ -12,6 +13,7 @@ def project_options(
         development_region: Optional. The development region for the project.
             Defaults to `"en"`.
         indent_width: Optional. The number of spaces to use for indentation.
+        organization_name: Optional. Populates the ORGANIZATIONNAME attribute for the project.
```
Comment (brentleyjones) [inline xcodeproj/internal/project_options.bzl:16]:
```suggestion
        organization_name: Optional. Populates the `ORGANIZATIONNAME` attribute
            for the project.
```

**Review by brentleyjones — COMMENTED** [review]
Missed some 80char wrapping I had in the previous comments.

**Pull request merged**